File: client/src/app/models/jour-ferie.ts
import { StatutAbsence } from "./statut-absence";
import { TypeJourFerie } from "./type-jour-ferie";

export interface JourFerie
{
  id: number;
  date: Date;
  libelle: string;
  type: TypeJourFerie;
  statut: StatutAbsence;
}

File: server/src/main/java/gestion_conges/server/entities/JourFerie.java
package gestion_conges.server.entities;

import gestion_conges.server.enums.TypeJourFerieEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Getter @Setter @Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
public class JourFerie
{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Column(nullable = false)
    private LocalDate date;
    @Column(nullable = false)
    private String libelle;
    @ManyToOne
    private TypeJourFerie type;
}
